Say hello to kddeployqt

bog-dan-ro opened the pull request.

bog-dan-kdab committed:
Say hello to kddeployqt tool

kddeployqt is a deployment tool for Qt applications, similar to (android/mac/win)deployqt which adds linux support.

bog-dan-kdab committed:
Add --xcb param

Adds xcbglintegrations which pulls the necessary X* libs

suy commented on README.md:
```diff
@@ -1,2 +1,9 @@
-Deployment tool for Qt applications, similar to (android/mac/win)deployqt
+Deployment tool for Qt applications, similar to (android/mac/win)deployqt.
+
+The main differences between this tool and the existing ones([linuxdeploqt](https://github.com/probonopd/linuxdeployqt), and [linuxdeploy](https://github.com/probonopd/linuxdeploy)) are:
```
I think the current, most popular tool named "linuxdeploy" is this: https://github.com/linuxdeploy/linuxdeploy

frankosterfeld commented on kddeployqt.py:
```diff
@@ -0,0 +1,274 @@
+            from_path = os.path.join(plugins_dir, plugin)
+            if (os.path.exists(from_path)):
+                if (args.verbose):
+                    print('\t\tcopying ' + from_path + ' to ' + out_plugins_dir)
```
I think format strings with placeholders would be more readable here and elsewhere.

pasnox commented on kddeployqt.py:
```diff
@@ -0,0 +1,278 @@
+                                print('Add ' + file + ' to the executables')
+                            executables.append(os.path.join(root, file))
+
+for plugin in args.extra_plugins:
+    copy_plugins([plugin])
```
why not just:
`copy_plugins(args.extra_plugins)`
?

bog-dan-kdab committed:
Add Qt5 support

bog-dan-kdab committed:
Say hello to kd_generate_deploy_app_script function

kd_generate_deploy_app_script is similar to qt_generate_deploy_app_script but with different parameters:
kd_generate_deploy_app_script (
     TARGET <target>
     OUTPUT_SCRIPT <output_script>
     [PROJECT_QML_PATHS <path> ...]
     [QML_IMPORT_PATHS <path> ...]
     [EXTRA_LIBS <path> ...]
     [EXTRA_PLUGINS <path> ...]
     [SYSTEM_LIBS_PATHS <path> ...]
     [WAYLAND]
     [XCB]
     [VERBOSE]
 )

dfaure-kdab commented on README.md:
```diff
@@ -1,2 +1,61 @@
+
+ ## How to use kddeployqt in your qt6 cmake project
+
+Adding kddeployqt support to your qt6 cmake project very easy:
```
*is* very easy

dfaure-kdab commented on README.md:
````diff
@@ -1,2 +1,61 @@
+)
+install(SCRIPT ${deploy_script})
+```
+Yup, is that easy ;-) !
````
*it* is that easy

dfaure-kdab commented on kddeployqt.py:
```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
```
MIT license header?

bog-dan-ro replied:
We have the license file, I don't think we need to pollute the src file with licenses.

dfaure-kdab commented on kddeployqt.py:
```diff
@@ -0,0 +1,308 @@
+    shutil.copy(args.executable, os.path.join(args.output, args.bin))
+
+# plugins dictionary
+# usesful until we can get the plugins info from libs folder as we do on android
```
typo: useful

dfaure-kdab commented on README.md:
```diff
@@ -1,2 +1,61 @@
-Deployment tool for Qt applications, similar to (android/mac/win)deployqt
+Deployment tool for Qt applications, similar to (android/mac/win)deployqt.
+
+The main differences between this tool and the existing ones([linuxdeploqt](https://github.com/probonopd/linuxdeployqt), [linuxdeploy](https://github.com/linuxdeploy/linuxdeploy), [linuxdeploy](https://github.com/probonopd/linuxdeploy), etc.) are:
```
Typo: linuxdeploqt -> linuxdeployqt
(missing 'y')

dfaure-kdab commented on README.md:
```diff
@@ -1,2 +1,61 @@
+
+The main differences between this tool and the existing ones([linuxdeploqt](https://github.com/probonopd/linuxdeployqt), [linuxdeploy](https://github.com/linuxdeploy/linuxdeploy), [linuxdeploy](https://github.com/probonopd/linuxdeploy), etc.) are:
+
+ - currently only linux host is supported.
```
This doesn't seem like a main difference with other tools called linuxdeploysomething, obviously linux only too.

At least move it last in your list, so that it's less obvious that you're not really listing the differences with the other tools, but more describing this one ;)

An real list of differences (maybe with a features table) would of course be more interesting to the reader, trying to find out which tool to use.

dfaure-kdab commented on README.md:
````diff
@@ -1,2 +1,61 @@
+
+```
+include(FetchContent)
+FetchContent_Declare(kddepoyqt
````
typo, missing 'l' in kddeployqt (same on line 22)

dfaure-kdab commented on CMakeLists.txt:
```diff
@@ -0,0 +1,165 @@
+# Usage:
+#
+# include(FetchContent)
+# FetchContent_Declare(kddepoyqt
```
depoy -> deploy here too (and on line 24)
